### DISCUSSION TITLE
How tu use @app in an external file?

### DISCUSSION THREAD

**Discussion by TiTi** [apiflask/apiflask#376] [Q&A] [answered]
Hello,

I'm trying to convert my Flask app (web app + api app) to APIFlask.

I've got it running locally, but the "@app." decorator **forces me to put everything in a single file**.
Any tip on how to avoid that?

My code is already using class-based views & models.
I've seen the related documentation : https://apiflask.com/usage/#use-class-based-views
I'm using exactly that, but putting the code in another file
See: https://github.com/LyonParapente/EventOrganizer/blob/main/back/app_flask.py#L145 and content of /back/api/ and /back/models/ (i'm rewritting them with marshmallow of course, code not pushed yet)

However, i can't use any decorator such as @app.input() or @app.output() this way because @app is not available in back/api/*.py
And if i import it from app_flask i get a circular import of course...

It's a code structure issue but I'm not a python guru
Any help appreciated !

**Comment by greyli** [comment, marked as answer]
You can use Blueprint to modular your app, check the docs here: https://flask.palletsprojects.com/en/2.2.x/blueprints/

(Be sure to replace `flask.Blueprint` with `apiflask.APIBlueprint` when changing to APIFlask.)

**Comment by TiTi** [comment]
Hey cool!
Thank you for your answer!

I wasn't used to blueprint before and didn't see the need/didn't really understand them.

I've made a POC and it's solves my problem. (now working on a proper commit :p)
I was stuck, thank you very much !

**Reply by greyli** [reply]
It's my pleasure :)
